lib/recommendation-server.js

```diff
@@ -71,14 +106,21 @@ RecommendationServer.prototype = {
     }
   },
   search: function(payload) {
+    let server = this;
```

Comment:
The binding issues here are just due to not assigning the bound `_resultHasChanged` back onto the object (see my comment at line 35). Once you make that change, you can get rid of the `server = this` assignment and update the code below. The arrow func preserves `this` binding, so it should just work. I'm happy to help debug if it continues to fight you :-)